components/net_traits/lib.rs

```diff
@@ -218,7 +220,13 @@ impl FetchTaskTarget for IpcSender<FetchResponseMsg> {
     }
 
     fn process_response(&mut self, response: &Response) {
-        let _ = self.send(FetchResponseMsg::ProcessResponse(response.metadata()));
+        let actual_metadata = if response.internal_response.is_some() && response.return_internal.get() {
```

Comment:
With my proposed design, we could restore the original code here and make `response.metadata()` return more meaningful data instead.